ruby-challenges.rb

```diff
@@ -3,30 +3,170 @@
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns an array of all the words containing that particular letter. Use the test variables provided.
 
+# PSEUDOCODE:
+
+# Given array and variables:
+# Input: ['coffee', 'tea', 'juice', 'water', 'soda water'] ("takes in..")
+
+# Input: 'o'
+# Output: ['coffee', 'soda water']
+
+# Input: 't'
+# Output: ['tea', 'water', 'soda water']
+
+# PROCESS:
+# Create/define a method (def) named "words_containing" that takes in an array of given words and a single letter in the parameters named "arrays, letters" and return an array of all the words containing that particular letter.
+# Create a new variable for the array named "new_array" to equal an array.
+# Since we are comparing whether a word has a specific letter in it, we need to use the dot notation select method (.select) to tell it to find and select a certain letter, then since we will need to get more information passed, we will use the ruby block "do" function to pass more information to the method, and then to determine the output for the value that we want we will use a parameter inside of pikes (||) to target each value and inside of the parameter pikes we will pass in the argument of string because we need to look at each element of the string in the array, and be sure to close the "do" block with an end to contain the logic. 
+# Next, we need to use a conditional statement (if) to invoke the string value and evaluate if the element of a string contains or includes certain letters that is given using the dot notation include method (.include) followed by a parameter to pass in the argument of "letters", and it will need to return that string with that letter only. At the end of the conditional statment, we will need to enter and "end" to close the code block.
+# Return an "end" to close the method code block.
+# We can print (p) out a statement by calling on the original method ('words_containing') together with the given variables passed in parameters (beverages_array, letter_o) and (beverages_array, letter_t) that will output the arrays of words that contain an "o" or "t".
+
 beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
 letter_o = 'o'
 # Expected output: ['coffee', 'soda water']
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def words_containing arrays, letters
+  new_array = arrays.select do |string|
+    if string.include?(letters)
+     string
```

Comment:
Looks good! Couple notes: no need to make the `new_array` variable at all. Also, the `.select` method intrinsically has an if statement, so no need to add one yourself.